app/src/renderer/components/staking/PageBond.vue

```diff
@@ -84,88 +138,76 @@ export default {
     FormGroup,
     FormMsg,
     FormStruct,
-    ListItem,
     Page,
     Part,
     ToolBar
   },
   computed: {
     ...mapGetters(['shoppingCart', 'user', 'committedDelegations']),
-    committedBondedAtoms () {
+    totalAtoms () {
+      return this.user.atoms + this.oldBondedAtoms
+    },
+    oldBondedAtoms () {
       return Object.values(this.committedDelegations).reduce((sum, d) => sum + d, 0)
     },
-    willUnbondAtoms () {
-      let sum = 0
-      /* eslint-disable no-unused-vars */
-      for (let [k, selectedDelegate] of this.fields.delegates.entries()) {
-        // set previously committed delegations for each delegate in cart
-        let previouslyCommitted = this.committedDelegations[selectedDelegate.id]
-
-        // check to see if user has allocated any atoms in cart
-        let currentlyAllocatedAtoms = selectedDelegate ? selectedDelegate.atoms : 0
-
-        // amount user intends to unbond from each delegate in cart
-        let unbondAmount = Math.max(previouslyCommitted - currentlyAllocatedAtoms, 0)
-
-        // set NaN's to 0
-        unbondAmount = unbondAmount || 0
-
-        // total amount user intends to unbond from all delegates in cart
-        sum += unbondAmount
-      }
-      return sum
+    oldUnbondedAtoms () {
+      return this.totalAtoms - this.oldBondedAtoms
+    },
+    newBondedAtoms () {
+      return this.fields.delegates.reduce((sum, d) => sum + (d.atoms || 0), 0)
     },
-    unbondedAtoms () {
-      let willBondSum = this.uncommittedBondedAtoms
-      let bondedSum = this.committedBondedAtoms
-      return this.totalAtoms - willBondSum + bondedSum - this.willUnbondAtoms
+    newUnbondedAtoms () {
+      let value = this.oldUnbondedAtoms
+      if (this.fields.delegates.length > 0) {
+        this.fields.delegates.map(d => {
```

Comment:
Maybe consider a reduce over mutation?
```
return this.fields.delegates.reduce((atoms, d) => {
  let delta = d.oldAtoms - d.atoms
  if (delta < 0) {
    return atoms + delta
  }
  return atoms
}, this.oldUnbondedAtoms)
```
But not important to me.

Reply:
Yep, looks cleaner!